src/annotator/host.coffee

```diff
@@ -4,6 +4,24 @@ Guest = require('./guest')
 
 module.exports = class Host extends Guest
   constructor: (element, options) ->
+
+    # Some options' values are not JSON-stringifiable (e.g. JavaScript
+    # functions) and will be omitted when the options are JSON-stringified.
+    # Add a JSON-stringifiable option for each of these so that the sidebar can
+    # at least know whether the callback functions were provided or not.
+    if options.services?[0]
+      service = options.services[0]
+      if service.onLoginRequest
+        service.onLoginRequestProvided = true
+      if service.onLogoutRequest
+        service.onLogoutRequestProvided = true
+      if service.onSignupRequest
+        service.onSignupRequestProvided = true
+      if service.onProfileRequest
+        service.onProfileRequestProvided = true
+      if service.onHelpRequest
+        service.onHelpRequestProvided = true
```

Comment:
Thanks for the explanation. I'm still not convinced that having settings being mapped from functions in the annotator code to booleans in the sidebar app code would be confusing. I'm OK to leave it as-is for the moment.